dashboard/test/controllers/scripts_controller_test.rb

```diff
@@ -88,11 +88,12 @@ class ScriptsControllerTest < ActionController::TestCase
     unit = create :script, published_state: nil, family_name: 'my-script'
     create :unit_group_unit, unit_group: course, script: unit, position: 1
 
+    @request.path = "/courses/#{course.name}/units/1"
     get :show, params: {
       id: unit.name,
     }
```

Comment:
if you're just testing what happens when you go to /courses/.../units/1 here, I would recommend passing these params directly:
```suggestion
        get :show, params: {course_course_name: course.name, position: 1}
```
here and below.